janitor/rabbitmq/publisher.py

```diff
@@ -0,0 +1,73 @@
+import json
+from io import StringIO
+from typing import Any, Dict, Optional, Sequence, cast
+
+from fastavro import json_writer, parse_schema
+from fastavro.types import Schema
+from pika import BasicProperties, BlockingConnection
+
+
+class Publisher:
+    """
+    Basic publisher class for sending messages to RabbitMQ.
+    """
```

Comment:
There is a library that Stuart developed that could had been reused for publishing into rabbitmq:

https://github.com/sanger/lab-share-lib/tree/develop/examples

Here is an example of code using the library:

https://github.com/sanger/lab-share-lib/blob/develop/examples/example1/publisher_example.py

Would it be possible to use it instead? That would save some duplicated functionality.

Reply:
I think it will be possible to use it, however, when publishing messages and there is an error in the process, my code will return the last message it tried to send and will retry next time the job is run. I'm not sure how to achieve the same thing using the library Stuart developed which seems to just log an error message after a certain number of retries.

Reply:
Ok